config/environments/development.rb

```diff
@@ -58,4 +58,12 @@
   # Use an evented file watcher to asynchronously detect changes in source code,
   # routes, locales, etc. This feature depends on the listen gem.
   config.file_watcher = ActiveSupport::EventedFileUpdateChecker
+
+  config.action_mailer.smtp_settings = {
+    user_name: Rails.application.credentials.dig(:development, :smtp_user_name),
+    password: Rails.application.credentials.dig(:development, :smtp_password),
```

Comment:
I had to use credentials.dig since credentials file is not divided by environments as secrets file. 
https://github.com/rails/rails/pull/30067